src/chapters/setup-intro-data-types/more-data-types.rst

```diff
@@ -2,258 +2,321 @@ More Data Types
+^^^^^^^^^^^^^^^^^^^^^^^
 
-We won’t explore every new aspect of this example, but rather will focus
-on the two methods.
+Let’s examine two classes in Java to explore defining and using methods. The
+first class is defined in a ``HelloMethods.java`` file, and it will have a
+``main`` method. The second class is defined in a separate ``Message.java``
+file, and it will contain a ``getMessage`` method that we want to call from
+within ``main``.
 
-.. raw:: html
+.. admonition:: Examples
 
-   <aside class="aside-warning">
+   ``HelloMethods.java``:
 
-As you’ve been following along with these examples using the code in
-IntelliJ, you’ve probably noticed that each class file, for example
-``Message.java`` and ``HelloMethods.java``, is named exactly the same as
-the class that file holds, for example ``Message`` and ``HelloMethods``
-respectively.
+   .. sourcecode:: java
+      :linenos:
 
-It is a rule in Java that a file containing a class marked ``public``
-**must** be named the same as that class. So remember to name each Java
-file you create to match the public class that file contains.
+      public class HelloMethods {
 
-.. raw:: html
+         public static void main(String[] args) {
+            String message = Message.getMessage("fr");
+            System.out.println(message);
+         }
 
-   </aside>
+      }
 
-The ``main`` method in the ``HelloMethods`` class is the same in
-structure as that of our previous examples. Take a look at the
-``Message`` class. Note that it *does not* have a ``main`` method, so it
-can’t be run on it’s own. Code within the ``Message`` class must be
-called from elsewhere in order to execute.
+   ``Message.java``:
 
-The ``Message`` class has a method of its own: ``getMessage``. Like
-``main``, it has the ``static`` keyword. Unlike ``main``, ``getMessage``
-has a return type of ``String``. It also has a single parameter,
-``String lang``.
+   .. sourcecode:: java
+      :linenos:
 
-Since Java is statically typed, each method must declare its return type
-– that is, the data type of what it will return – along with the type of
-each parameter. One consequence of this that may not be immediately
-obvious is that a method in Java may not have return statements that
-return different types of data. For example, we would not be able to
-replace the last ``return`` statement of ``getMessage`` with something
-like ``return 42;``. This would be flagged as a compiler error.
+      public class Message {
+
+         public static String getMessage(String lang) {
+
+            if (lang.equals("sp")) {
+               return "¡Hola, Mundo!";
+            } else if (lang.equals("fr")) {
+               return "Bonjour, le monde!";
+            } else {
+               return "Hello, World!";
+            }
+         }
+      }
+
+We won’t explore every new aspect of this example, but instead focus on the two
+methods.
+
+   TODO: Update local references after we decide which code-comparison page
```

Comment:
you can remove the older code-comparison page

Reply:
Done.